Sources/DeepLearning/Optimizer.swift

```diff
@@ -233,25 +243,34 @@ public class SGD<Model: Layer, Scalar: TensorFlowFloatingPoint>: Optimizer
                 model[keyPath: kp] += velocity[keyPath: kp]
             }
         }
+        for kp in model.recursivelyAllWritableKeyPaths(to: Tensor<Double>.self) {
+            velocity[keyPath: kp] =
+                Double(momentum) * velocity[keyPath: kp] - Double(learningRate) * direction[keyPath: kp]
+            if nesterov {
+                model[keyPath: kp] +=
+                    Double(momentum) * velocity[keyPath: kp] - Double(learningRate) * direction[keyPath: kp]
+            } else {
+                model[keyPath: kp] += velocity[keyPath: kp]
+            }
+        }
     }
 }
 
 // MARK: - Manifold optimizers
 
 /// A Riemann manifold stochastic gradient descent (SGD) optimizer.
-public class RiemannSGD<Model: Layer, Scalar: FloatingPoint>: Optimizer
```

Comment:
Let's not change the `Riemann` one for now since it's not `Tensor`-specific and no one is using it.

Reply:
Reverted changes.